src/layer/x86/instancenorm_x86.cpp

```diff
@@ -0,0 +1,191 @@
+        }
+
+        i = 0;
+
+        float mean = sum / size;
+        float tmp = 0.f;
+
+#if __SSE2__
+#if __AVX__
+#if __AVX512F__
+        __m512 _sqsum_avx512 = _mm512_setzero_ps();
+        __m512 _mean_avx512 = _mm512_set1_ps(mean);
+
+        for (; i + 15 < size; i += 16)
+        {
+            __m512 _p = _mm512_loadu_ps(ptr + i);
+            __m512 _diff = _mm512_sub_ps(_p, _mean_avx512);
+            _sqsum_avx512 = _mm512_fmadd_ps(_diff, _diff, _sqsum_avx512);
+        }
+        sqsum += _mm512_comp_reduce_add_ps(_sqsum_avx512);
+#endif // __AVX512F
+        __m256 _sqsum_avx = _mm256_setzero_ps();
+        __m256 _mean_avx = _mm256_set1_ps(mean);
+
+        for (; i + 7 < size; i += 8)
+        {
+            __m256 _p = _mm256_loadu_ps(ptr + i);
+            __m256 _diff = _mm256_sub_ps(_p, _mean_avx);
+            _sqsum_avx = _mm256_comp_fmadd_ps(_diff, _diff, _sqsum_avx);
+        }
+        sqsum += _mm256_reduce_add_ps(_sqsum_avx);
+#endif // __AVX__
+        __m128 _sqsum = _mm_setzero_ps();
+        __m128 _mean = _mm_set1_ps(mean);
+
+        for (; i + 3 < size; i += 4)
+        {
+            __m128 _p = _mm_loadu_ps(ptr + i);
+            __m128 _diff = _mm_sub_ps(_p, _mean);
+            _sqsum = _mm_comp_fmadd_ps(_diff, _diff, _sqsum);
+        }
+        sqsum += _mm_reduce_add_ps(_sqsum);
+#endif // __SSE2__
+
+        for (; i < size; i++)
+        {
+            tmp = ptr[i] - mean;
+            sqsum += tmp * tmp;
+        }
+        float var = sqsum / size;
+        // the var maybe minus due to accuracy
+        //float var = sqsum / size - mean * mean;
+
+        float a;
+        float b;
+        if (affine)
+        {
+            float gamma = gamma_data[q];
+            float beta = beta_data[q];
+
+            a = gamma / (sqrtf(var + eps));
+            b = -mean * a + beta;
+        }
+        else
+        {
+            a = 1.f / (sqrtf(var + eps));
+            b = -mean * a;
+        }
+
+        i = 0;
+
+#if __SSE2__
+#if __AVX__
+#if __AVX512F__
+        __m512 _va_avx512 = _mm512_set1_ps(a);
+        __m512 _vb_avx512 = _mm512_set1_ps(b);
+
+        for (; i + 16 < size; i += 16)
+        {
+            __m512 _p = _mm512_loadu_ps(ptr + i);
+            _p = _mm512_fmadd_ps(_p, _va_avx512, _vb_avx512);
+            _mm512_storeu_ps(ptr + i, _p);
+        }
+#endif // __AVX512F__
+        __m256 _va_avx = _mm256_set1_ps(a);
+        __m256 _vb_vax = _mm256_set1_ps(b);
+
+        for (; i + 7 < size; i += 8)
+        {
+            __m256 _p = _mm256_loadu_ps(ptr + i);
+            _p = _mm256_comp_fmadd_ps(_p, _va_avx, _vb_vax);
+            _mm256_storeu_ps(ptr + i, _p);
+        }
+#endif // __AVX__
+        __m128 _va = _mm_set1_ps(a);
+        __m128 _vb = _mm_set1_ps(b);
+
+        for (; i + 3 < size; i += 4)
+        {
+            __m128 _p = _mm_loadu_ps(ptr + i);
+            _p = _mm_comp_fmadd_ps(_p, _va, _vb); // Fused Multiply-Add
+            _mm_storeu_ps(ptr + i, _p);
```

Comment:
`ptr + i` ➡️ `ptr` and `ptr += 4` at the loop end

code style ref https://github.com/Tencent/ncnn/blob/master/src/layer/x86/layernorm_x86.cpp